lib/internal/modules/esm/hooks.js

```diff
@@ -84,7 +84,7 @@ class Hooks {
   };
 
   // Enable an optimization in ESMLoader.getModuleJob
-  hasCustomLoadHooks = false;
+  hasCustomResolveOrLoadHooks = false;
```

Comment:
why the re-name?

Reply:
This was previously a flag to tell `getModuleJob` that a custom `load` hook was registered (it didn’t need to know about any other hooks). Now it needs to know about custom `load` and `resolve` hooks, hence setting this for `resolve` too and renaming it.

This disappears in #44710.